File: src/Java/remove_duplicate_from_sorted_array.java

class RemoveDuplicates {
    public int removeDuplicates(int[] nums) {
        int index=0;

        for (int i = 0; i < nums.length-1; i++) {
            if (nums[i] < nums[i+1]) {
                nums[index] = nums[i];
                index++;
            }
            if(i+1 == nums.length && nums[i] < nums[i+1]){
                index++;
            }
        }

        return index;
    }

    public void main(String[] args) {
        System.out.println(removeDuplicates(new int[]{0,0,1,1,1,2,2,3,3,4}));
        System.out.println("hi");
    }
}

File: src/Python/remove_duplicate_from_sorted_array.py
class Solution:
    def removeDuplicates(self, nums: list[int]) -> int:
        index = 1

        for i in range(len(nums)-1):
            if nums[i+1] > nums[i]:
                nums[index] = nums[i+1]
                index+=1

        return index

        
solution = Solution()
print(solution.removeDuplicates([1,1,2]))
print(solution.removeDuplicates([0,0,1,1,1,2,2,3,3,4]))